staging/src/k8s.io/apimachinery/pkg/conversion/queryparams/convert.go

```diff
@@ -115,14 +115,17 @@ func customMarshalValue(value reflect.Value) (reflect.Value, bool) {
 }
 
 func addParam(values url.Values, tag string, omitempty bool, value reflect.Value) {
-	if omitempty && zeroValue(value) {
```

Comment:
zeroValue should not return true for a *int64 set to 0. what value was getting passed in here?

Reply:
It's `tailLines`, see #35335 
If user set that value to zero, the program should not ignore it, cause he may want to tail 0 lines.

Reply:
Maybe the problem is not here...

https://play.golang.org/p/05D2ZcWVr7i

Just remove the `.Elem()` might works but I have not test it yet: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/conversion/queryparams/convert.go#L147

Reply:
https://github.com/kubernetes/kubernetes/blob/20934328ce1357073d53e922275204f76488b460/staging/src/k8s.io/apimachinery/pkg/conversion/queryparams/convert.go#L176

In that case, where we’ve determined we have a pointer type that is non-nil, when we evolve the actual value to serialize, we should always set omitempty to false

Reply:
Agree, so the changes will be:

```go
if isPointerKind(kind) {
	ft = ft.Elem()
	kind = ft.Kind()
	if !field.IsNil() {
		field = reflect.Indirect(field)
		omitempty = false
	}
}
```

Is that OK?

Reply:
That seems right. Would like a test demonstrating the serialization that fails currently so we can make sure this fixes the problem

Reply:
I'll add more unit-tests for that later.